packages/predicate/src/predicate.ts

```diff
@@ -36,4 +46,36 @@ export class Predicate extends AbstractPredicate {
       }
     }
   }
+
+  populateTransactionPredicateData(transactionRequestLike: TransactionRequestLike) {
+    const request = transactionRequestify(transactionRequestLike);
+
+    request.inputs?.forEach((input) => {
+      if (input.type === InputType.Coin && hexlify(input.owner) === this.address.toB256()) {
+        // eslint-disable-next-line no-param-reassign
+        input.predicate = this.bytes;
+        // eslint-disable-next-line no-param-reassign
+        input.predicateData = this.predicateData;
+      }
+    });
+
```

Comment:
can we use `request.inputs?.map` instead to avoid reassigning `forEach` params ?

Reply:
The idea here is to reassign the params to inject the new data in the tx, otherwise, it will create new inputs and would need to replace the entire inputs section.